src/gt4py/next/ffront/past_to_itir.py

```diff
@@ -247,7 +255,9 @@ def _construct_itir_domain_arg(
         else:
             raise AssertionError()
 
-        return itir.FunCall(fun=itir.SymRef(id=domain_builtin), args=domain_args)
+        return itir.FunCall(
+            fun=itir.SymRef(id=domain_builtin), args=domain_args, location=out_field.location
```

Comment:
Such that the location of the `node_domain` is used when available.
```suggestion
            fun=itir.SymRef(id=domain_builtin), args=domain_args, location=(node_domain or out_field).location
```